engine/apps/alerts/models/alert_group.py

```diff
@@ -401,15 +401,21 @@ def is_alert_a_resolve_signal(self, alert):
         raise NotImplementedError
 
     @property
-    def permalink(self):
+    def slack_permalink(self):
         if self.slack_message is not None:
             return self.slack_message.permalink
 
+    @property
+    def telegram_permalink(self) -> typing.Optional[str]:
+        if hasattr(self, "prefetched_telegram_messages"):
+            return self.prefetched_telegram_messages[0].link if self.prefetched_telegram_messages else None
+        return None
```

Comment:
In case it's not prefetched I think we should make a query in this method
```suggestion
        return self.telegram_messages.filter(chat_id__startswith="-", message_type=TelegramMessage.ALERT_GROUP_MESSAGE).first()

```